pkg/tsdb/prometheus/prometheus.go

```diff
@@ -81,6 +91,45 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 	return i.buffered.ExecuteTimeSeriesQuery(ctx, req)
 }
 
+func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	if !strings.HasPrefix(req.URL, "/api/v1") && !strings.HasPrefix(req.URL, "api/v1") {
+		return sender.Send(&backend.CallResourceResponse{
+			Status: 404,
+			Headers: map[string][]string{
+				"content-type": {"application/json"},
+			},
+			Body: []byte(fmt.Sprintf("%s: Not a valid prometheus URL", req.URL)),
+		})
+	}
+
+	i, err := s.getInstance(req.PluginContext)
+	if err != nil {
+		return err
+	}
+
+	statusCode, bytes, err := i.resource.Execute(ctx, req)
+	if statusCode >= 300 {
```

Comment:
i'd suggest also checking the error here:
```suggestion
	if statusCode >= 300 || err != nil {
```